### DISCUSSION TITLE
How to run c++ with embedded python inside on any version of python installed on the system

### DISCUSSION THREAD

**Discussion by CncGpp** [pybind/pybind11#3428] [Q&A]
Hi, I build a simple CLI executable which can run small python scripts.  All work great on my machine w python38 but i cant distribuite it becouse it's strictly bindend to the my current python version (3.8.x).

- Is in somehow possible to run the executable on whatever python 3.X version is find? Mybe dynamically load the found dll?  NB I dont wont to statically link python inside my app, but just use the version aveabile on the machine.  
- For now I'm getting around the problem by by placing it in the app folder the python38.dll & python38.zip (the embeddable from the official site) How can i set the ENV variable to allow to put those file in another dir?

Downloading any of embeddable version (3.7, 3.8, 3.9, 3.10) and simply renaming the .dll e the .zip python38.dll & python38.zip also work without problem, the compatibility (at least for this case) is not a problem.

Thx

**Comment by zliucd** [comment]
Python is interpreter-based, a possible workaround is compile different C++ executables/libraries corresponding to Python versions. As there are only a few Python mainstream versions(3.6, 3.8 etc),  it should work.

**Reply by CncGpp** [reply]
Yes it is a solution, but I was hoping for something more "transparent" or some workaround to use the installed interpreter; in many cases the end user does not even know he have python or it is not practical.

Perhaps the best solution is to create an installer that carries the right embed version of python with it.
